lib/clientv2/interface.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientv2
+
+import "github.com/projectcalico/libcalico-go/lib/ipam"
+
+type Interface interface {
```

Comment:
an interface called `Interface`? How about `ClientInterface` or something similar? Same for the filename

Reply:
actually this is common practice in k8s and I quite like it.  The Interface represents the primary interface for the package.  Sub-interfaces do need additional naming, but having the main entry point as just Interface is quite nice.  Thus a caller to the code uses client.Interface.

client.ClientInterface is generally discouraged due to repetition.